### PR TITLE
Re-add some deps and install -dev libs correctly

### PR DESCRIPTION
**Pull request by mckees**

### CODE REVIEW

File: debian/control
Comment (androniychuk) [inline debian/control]: this might be libigsc and not intel-gsc.... need to check

**Comment by androniychuk** [comment]
also.... the deps should probably go into xpu-smi
as I don't know who will use the libxpu without xpu-smi

File: debian/control
Comment (mckees) [reply debian/control]:
```
$ ldd /usr/bin/xpu-smi 
	linux-vdso.so.1 (0x00007d157db47000)
	*libigsc.so.0 => /lib/x86_64-linux-gnu/libigsc.so.0 (0x00007d157d99f000)*
	libxpum.so.1 => /lib/x86_64-linux-gnu/libxpum.so.1 (0x00007d157d400000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007d157d000000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007d157d8b2000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007d157d884000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007d157cc00000)
	libudev.so.1 => /lib/x86_64-linux-gnu/libudev.so.1 (0x00007d157d83d000)
	libze_loader.so.1 => /lib/x86_64-linux-gnu/libze_loader.so.1 (0x00007d157d33b000)
	libpciaccess.so.0 => /lib/x86_64-linux-gnu/libpciaccess.so.0 (0x00007d157d832000)
	/lib64/ld-linux-x86-64.so.2 (0x00007d157db49000)
	libcap.so.2 => /lib/x86_64-linux-gnu/libcap.so.2 (0x00007d157d825000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007d157d805000)
```

Looks like that's correct. I'll update it

**Comment by mckees** [comment]
> also.... the deps should probably go into xpu-smi as I don't know who will use the libxpu without xpu-smi

I don't think this makes sense from a dependency point of view though. We want the most independent package to have the deps it needs. Then those deps carry forward to any other package that depends on it. Since libxpum1 doesn't depend on the other two, it should have the deps, then xpu-smi only needs a dep on libxpum1

File: debian/control
Comment (hector-cao) [inline debian/control:27]: @mckees I dont think we need to explicitly specify these dependencies, the helper `${shlibs:Depends}` will automatically detect them for us

Comment (hector-cao) [reply debian/control:27]: I checked and it automatically detects `libigsc0` and `libze1`, but not ` libze-intel-gpu1` because it is loaded at runtime (see `core/src/diagnostic/diagnostic_manager.cpp`)

**Review by hector-cao — CHANGES_REQUESTED** [review]

File: debian/libxpum-dev.install
Comment (hector-cao) [inline debian/libxpum-dev.install]: Here are the contents of `usr/lib/xpu-smi` 

```
usr/lib/xpu-smi/
|-- config
|   |-- diagnostics.conf
|   |-- pci.conf
|   |-- pci.ids
|   |-- perf_metrics.conf
|   |-- vgpu.conf
|   |-- xpum.conf
|   `-- xpum.conf.template
`-- resources
    |-- amcfw
    |   `-- Intel\ Simplified\ Software\ License\ August\ 2021_Final_08.20.21_v2.txt
    |-- kernels
    |   |-- LICENSE
    |   |-- test_multiple_memory_allocations.cl
    |   |-- test_multiple_memory_allocations.spv
    |   |-- ze_dp_compute.cl
    |   |-- ze_dp_compute.spv
    |   |-- ze_global_bw.cl
    |   |-- ze_global_bw.spv
    |   |-- ze_int_compute.cl
    |   |-- ze_int_compute.spv
    |   |-- ze_sp_compute.cl
    |   `-- ze_sp_compute.spv
    `-- mediadata
        |-- LICENSE
        |-- test_stream.264
        |-- test_stream_1080p.265
        `-- test_stream_4K.265
```

I have some doubts about the fact that they are intended to be in the -dev package,

**Review by hector-cao — CHANGES_REQUESTED** [review]

File: debian/libxpum-dev.install
Comment (hector-cao) [reply debian/libxpum-dev.install]: @androniychuk do you know ?

File: debian/control
Comment (mckees) [reply debian/control:27]: When I had them removed before, I was able to install the package without these dependencies, which seems like it was something that Pavel was also seeing.

File: debian/libxpum-dev.install
Comment (mckees) [reply debian/libxpum-dev.install]: I'm open to removing those, but I've kept them in since they appear to be test resources, and I don't want to package tests that will fail without necessary resources. @androniychuk are these important for the package?

Comment (hector-cao) [reply debian/libxpum-dev.install]: I did not mean to remove them, just they do not seem to be in the right package, we have to check but they can be useful to `xpu-smi` or even `libxxpum1`

File: debian/control
Comment (hector-cao) [reply debian/control:27]: Weird, do we still have this package in the PPA ? I can take a look, when i build the package without these dependencies, i can have the `libxpum1` binary package with these dependencies

```
Depends: libze-intel-gpu1 (>= 1.3.23726), libc6 (>= 2.38), libgcc-s1 (>= 3.4), libigsc0 (>= 0.9.3), libpciaccess0, libstdc++6 (>= 14), libze1 (>= 1.17.42)
```

Comment (androniychuk) [reply debian/control:27]: version needs to be bumped for libze-intel-gpu1

File: debian/libxpum-dev.install
Comment (androniychuk) [reply debian/libxpum-dev.install]: everything in this dir, needs to be there
"usr/lib/xpu-smi"

File: debian/control
Comment (hector-cao) [reply debian/control:27]: @mckees I just checked from the testing PPA and see the behavior what you mentioned, let me try to understand

**Comment by androniychuk** [comment]
> > also.... the deps should probably go into xpu-smi as I don't know who will use the libxpu without xpu-smi
> 
> I don't think this makes sense from a dependency point of view though. We want the most independent package to have the deps it needs. Then those deps carry forward to any other package that depends on it. Since libxpum1 doesn't depend on the other two, it should have the deps, then xpu-smi only needs a dep on libxpum1

I understand what your saying, but consider this:

the library libxpu....so can be used in different situations, theoretically .... 
but the program xpu-smi, should ONLY be run when those other pkgs are present in the system

**Comment by mckees** [comment]
> the library libxpu....so can be used in different situations, theoretically .... but the program xpu-smi, should ONLY be run when those other pkgs are present in the system

Yes, and with the current configuration, xpu-smi can only install if those libs are present plus libxpum1, which it also requires. So this is doing what you suggest but using libxpum1 as a way to inherit those deps

File: debian/control
Comment (androniychuk) [inline debian/control]: version needs to match at least the 24.35 version of libze-intel-qpu1...this is old version scheme

**Comment by androniychuk** [comment]
> > the library libxpu....so can be used in different situations, theoretically .... but the program xpu-smi, should ONLY be run when those other pkgs are present in the system
> 
> Yes, and with the current configuration, xpu-smi can only install if those libs are present plus libxpum1, which it also requires. So this is doing what you suggest but using libxpum1 as a way to inherit those deps

this is almost correct, because libxpum1 can theoretically be used in non-Intel systems, so therefore we should not force people to load those pkgs from that level

File: debian/libxpum-dev.install
Comment (androniychuk) [reply debian/libxpum-dev.install]: here's some relevant docs

https://github.com/intel/xpumanager/blob/master/doc/smi_user_guide.md#diagnose-gpu-with-different-test-suites

File: debian/control
Comment (hector-cao) [reply debian/control:27]: @mckees I think i understand now, the `{shlibs:Depends}` helper did not work because we are putting the shared library in the `-dev` package

File: debian/libxpum-dev.install
Comment (hector-cao) [reply debian/libxpum-dev.install]: @mckees my feeling is that we need to install this folder for the `xpu-smi` package, but we should double check

Comment (androniychuk) [reply debian/libxpum-dev.install]: I just realized that this is in the -dev pkg...
I totally agree it must be in xpu-smi pkg

File: debian/control
Comment (mckees) [reply debian/control]: Updated to the current version for now

File: debian/libxpum-dev.install
Comment (mckees) [reply debian/libxpum-dev.install]: Alright, I switched it to the xpu-smi package. Since it was under libs but wasn't part of the actual libs, I decided to put it in -dev

File: debian/xpu-smi.install
Comment (androniychuk) [inline debian/xpu-smi.install]: /usr/bin/xpu-smi

**Comment by mckees** [comment]
> this is almost correct, because libxpum1 can theoretically be used in non-Intel systems, so therefore we should not force people to load those pkgs from that level

Oh, I didn't realize that there was an intention to use the lib outside of Intel hardware but not the binary... Anyway, I updated it

File: debian/xpu-smi.install
Comment (mckees) [reply debian/xpu-smi.install]: The assumption being that any new binaries we add would be installed via a different package? That's currently the only file in usr/bin, so I just want to make sure before every new binary requires a new change to the deb install process

**Review by androniychuk — APPROVED** [review]

File: debian/control
Comment (hector-cao) [inline debian/control:40]: @mckees why do we want these dependencies here now ? are they direct dependencies of xpu-smi ?

**Review by hector-cao — CHANGES_REQUESTED** [review]

File: debian/control
Comment (mckees) [reply debian/control:40]: @androniychuk has mentioned that they are runtime dependencies for xpu-smi

Comment (androniychuk) [reply debian/control:40]: without those pkgs installed on an intel system ..
xpu-smi is useless
runtime dependencies...

Comment (hector-cao) [reply debian/control:40]: got it but are they direct dependencies or not ? if they are direct dependencies of libxpum, no need to specify them here since libxpum is already a dependency for xpu-smi and they will be all installed when we install xpu-smi,

File: debian/control
Comment (androniychuk) [inline debian/control]: I also recommend to move the "recommends" and "suggests" to xpu-smi

File: debian/control
Comment (androniychuk) [reply debian/control:40]: they are not necessarily build-time dependencies, but they are runtime dependencies, there's a difference there

Comment (hector-cao) [reply debian/control:47]: @androniychuk :) sorry but that does not answer my question : direct dependency (runtime or not, whatever) ?

**Review by hector-cao — CHANGES_REQUESTED** [review]

File: debian/control
Comment (hector-cao) [reply debian/control:40]: @androniychuk :) sorry but that does not answer my question : direct dependency (runtime or not, whatever) ?

Comment (androniychuk) [reply debian/control:40]: I feel like I answered it. can you give me a definition of what you mean by direct dependencies or something referencing it in the pkging manual?

File: debian/control
Comment (mckees) [reply debian/control]: Added a new commit to move those over

File: debian/libxpum-dev.install
Comment (androniychuk) [inline debian/libxpum-dev.install:1]: I was always told that it's important/better to include the individual header files instead of a the whole directory...
is this true?
```
./usr/include/xpum_api.h
./usr/include/xpum_structs.h
```

Comment (androniychuk) [reply debian/libxpum-dev.install:1]: or something like /usr/include/xpum_*.h
when there's lots of them

Comment (mckees) [reply debian/libxpum-dev.install:1]: Is the package installing a whole directory of unnecessary files? If we don't want them, we can remove them, but I'm more inclined to install everything to avoid unintentionally breaking some functionality. That leaves the upstream devs the ability to change what gets installed without everything getting broken

Comment (androniychuk) [reply debian/libxpum-dev.install:1]: its only those headers files in the directory, but from a pkging level it helps to see whats in the dev pkg

Comment (androniychuk) [reply debian/libxpum-dev.install:1]: the build will break if there r extra files and not specified anywhere

Comment (androniychuk) [reply debian/libxpum-dev.install:1]: either way, your call, the PR is ready to merge

**Review by androniychuk — APPROVED** [review]

File: debian/control
Comment (hector-cao) [reply debian/control:40]: That is what i have in mind :
```
A() {
  B()
}
B() {
  C()
}
B is direct dependency of A, C is not a direct dependency of A so C should not be expressed as A's dependency
```

```
A() {
  B()
  C()
}
B & C are direct dependencies of A
```

Does that make sense ?

Comment (androniychuk) [reply debian/control:40]: > without those pkgs installed on an intel system .. xpu-smi is useless runtime dependencies...

here is my previous response, this is a more detailed description of the dependencies
runtime dependencies, they need to be there in order for xpu-smi to work properly.
They are not dependencies of another pkg that is a dependency of xpu-smi.

Comment (hector-cao) [reply debian/control:40]: Seems that i cannot be clear enough to make you understand my point, i will not block the PR for that if @mckees is fine with that, i will approve

**Review by hector-cao — APPROVED** [review]
LGTM

**Pull request merged**
